tensorflow/compiler/mlir/tosa/transforms/strip_quant_types.cc

```diff
@@ -52,69 +55,28 @@ namespace {
 #define GEN_PASS_CLASSES
 #include "tensorflow/compiler/mlir/tosa/transforms/passes.h.inc"
 
-class StripQuantTypes : public TosaStripQuantTypesPassBase<StripQuantTypes> {
- public:
-  explicit StripQuantTypes() {}
-  void runOnFunction() override;
-};
-
-class QuantTypeConverter : public TypeConverter {
- public:
-  static Type convertType(Type type) {
-    if (auto qType = type.dyn_cast<quant::QuantizedType>()) {
-      if (qType.isSigned() || qType.getStorageTypeIntegralWidth() != 8) {
-        return IntegerType::get(type.getContext(),
-                                qType.getStorageTypeIntegralWidth());
-      }
+bool isOpTosa(mlir::Operation *op)
+{
+  return (isa<tosa::TosaDialect>(op->getDialect()));
+}
 
-      return IntegerType::get(type.getContext(),
-                              qType.getStorageTypeIntegralWidth(),
-                              IntegerType::SignednessSemantics::Unsigned);
-    }
-    return type;
-  }
-  static Type convertTensor(RankedTensorType type) {
-    auto newType = RankedTensorType::get(type.getShape(),
-                                         convertType(type.getElementType()));
-    return newType;
+Type convertType(Type type)
+{
+  if (auto qType = type.dyn_cast<quant::QuantizedType>()) {
+    return IntegerType::get(type.getContext(),
+                            qType.getStorageTypeIntegralWidth(),
+                            qType.isSigned() ? IntegerType::SignednessSemantics::Signed:
+                                               IntegerType::SignednessSemantics::Unsigned);
   }
-  explicit QuantTypeConverter() {
-    addConversion([](Type type) { return convertType(type); });
-    addConversion(convertTensor);
-  }
-};
-
-// Handles the type conversion component of the TypeConversion. This updates
-// conversion patterns that used the original Quant types to be updated to
-// the non-quant variants.
-class GenericTypeConvert : public ConversionPattern {
```

Comment:
If you want to conditionally convert TOSA operations, you should check the operation is a tosa dialect operation, perform the type stripping on the TOSA operation, then include UnrealizedConversion casts to maintain the quantization on types that use the result.

This same approach should be used on argument / return types.